setuptools/command/bdist_wheel.py

```diff
@@ -597,3 +595,18 @@ def adios(p: str) -> None:
             shutil.copy(license_path, os.path.join(distinfo_path, filename))
 
         adios(egginfo_path)
+
+    def _zip_compression(self) -> int:
+        if (
+            isinstance(self.compression, int)
+            and self.compression in self.supported_compressions.values()
+        ):
+            return self.compression
+
+        try:
+            if isinstance(self.compression, str):
+                return self.supported_compressions[self.compression]
+        except KeyError:
+            pass
```

Comment:
I had previously done

```suggestion
        compression = self.supported_compressions.get(self.compression)
        if compression is not None:
            return compression
```

To make the type-checker happy (even through this is wrapped in a try-except anyway). Which technically allows using something else than a str as long as it's stringified to a valid compression str. But that's extremely niche. Might be desired if that *feels* cleaner, up to you 🤷

Reply:
Ok, so we do need the `if` guard here according to the last result of the CI.